hudi-common/src/main/java/org/apache/hudi/common/table/view/HoodieTableFileSystemView.java

```diff
@@ -378,19 +383,45 @@ protected Option<Pair<String, CompactionOperation>> getPendingLogCompactionOpera
 
   @Override
   protected boolean isPartitionAvailableInStore(String partitionPath) {
-    return partitionToFileGroupsMap.containsKey(partitionPath);
+    fsLock.readLock();
+    try {
+      return partitionToFileGroupsMap.containsKey(partitionPath);
+    } finally {
+      fsLock.readUnlock();
+    }
   }
 
   @Override
   protected void storePartitionView(String partitionPath, List<HoodieFileGroup> fileGroups) {
-    LOG.debug("Adding file-groups for partition :" + partitionPath + ", #FileGroups=" + fileGroups.size());
-    List<HoodieFileGroup> newList = new ArrayList<>(fileGroups);
-    partitionToFileGroupsMap.put(partitionPath, newList);
+    fsLock.writeLock();
+    try {
+      LOG.debug("Adding file-groups for partition :" + partitionPath + ", #FileGroups=" + fileGroups.size());
+      List<HoodieFileGroup> newList = new ArrayList<>(fileGroups);
+      partitionToFileGroupsMap.put(partitionPath, newList);
+    } finally {
+      fsLock.writeUnlock();
+    }
+  }
+
+  @Override
+  protected void removePartitionView(String partitionPath) {
+    fsLock.writeLock();
```

Comment:
Same here - this is also being called alrady from the critical section in `cleanFileGroupsInPartitions`. Why take lock again?